source/branching.rst

```diff
@@ -0,0 +1,210 @@
+
+Feature branches should always derive from ``develop``.  ``develop`` is the stable version of the
+new code, and you must ensure that your feature will play nice with the current development state.
+In git, you can create a branch, switch to it, and pick its parent branch all at once with the command::
+
+    $ git checkout -b myfeature develop
+
+It is a good habit to branch this way, because developers often forget to checkout the branches they
+just created, and also may forget where they were before they cut a new branch. Both of these human
+errors might cause merge headaches later.  You are allowed to create a feature branch from another
+feature branch instead of ``develop``, if you are comfortable with the parent branch and have a reason
+to branch from it.  However, all feature branches must eventually merge back into ``develop`` or deleted.
+
+It's recommended practice to prefix the branch name with the issue number which it addresses. For example::
+
+    $ git checkout -b 42-my-new-feature develop
+
+Now that your branch is created, develop against it. Make as many commits as you'd like, and even
+as many pushes to origin as you'd like.  For example::
+
+    $ git add .
+    $ git commit -m "refactored models to address issue #42"
+    $ git push origin 42-my-new-feature -u
+
+Commit messages should be descriptive. Where practical, note changed files, why you changed them, and
+other metadata (WIP, not yet tested, addresses issue #42, etc.)
+
+
+Pull Requests
+-------------
+
+When you are certain of the following:
+
+- The branch has addressed the issue
+- All tests pass
+- Test coverage is above 80% for the entire repository
+- All code is PEP8 compliant
+- All migrations created if needed
+- All print/debug statements removed
+- The server runs as expected
+
+then you can do a final push and create a pull request.  GitHub makes this simple. There is a
+green button that reads "Review and create pull request."  You should look over the diff of the code
+and make sure it reads as you intend.  If so, add a comment to the pull request that it addresses issue #42.
+
+It is then incumbent on another member of the team to do the following:
+
+- Pull your feature branch
+- Run tests (this may be handled by a CI server such as TravisCI)
+- Run migrations (this may be handled by a CI server such as TravisCI)
+- Run their local dev server
+- Verify that the feature was addressed
+
+It is also incumbent upon the pull request reviewer to read the code diff and make comments on the code.
+Github presents a blue and white plus icon for adding line comments.
+
+When the reviewer agrees that the pull request is adequate, they should add a comment to the pull request.
+For fun, we typically add random emoticons such as ``:frog:``.
+
+By convention, the developer of the branch then merges it back into ``develop``, but other developers can do this step::
+
+    $ git checkout develop
+    $ git merge --no-ff 42-my-new-feature
```

Comment:
I think it is worth mentioning the `--no-ff` flag and the reason its recommended.